reVX/setbacks/solar_setbacks.py

```diff
@@ -0,0 +1,219 @@
+# -*- coding: utf-8 -*-
+"""
+Compute setbacks exclusions
+"""
+import logging
+import os
+import geopandas as gpd
+
+from rex.utilities import log_mem
+
+from reVX.setbacks.base import BaseSetbacks
+
+
+logger = logging.getLogger(__name__)
+
+
+class ParcelSetbacks(BaseSetbacks):
+    """
+    Parcel setbacks, using negative buffers.
+    """
+
+    def compute_generic_setbacks(self, features_fpath):
+        """Compute generic setbacks.
+
+        Parameters
+        ----------
+        features_fpath : str
+            Path to shape file with features to compute setbacks from.
+
+        Returns
+        -------
+        setbacks : ndarray
+            Raster array of setbacks
+        """
+        logger.info('Computing generic setbacks')
+        setback_features = self._parse_features(features_fpath)
+
+        setbacks = [
+            (geom, 1) for geom in setback_features.buffer(0).difference(
+                setback_features.buffer(-1 * self.generic_setback)
+            )
+        ]
+
+        return self._rasterize_setbacks(setbacks)
+
+    @staticmethod
+    def _compute_local_setbacks(features, cnty, setback):
+        """Compute local features setbacks.
+
+        Parameters
+        ----------
+        features : geopandas.GeoDataFrame
+            Features to setback from.
+        cnty : geopandas.GeoDataFrame
+            Regulations for a single county.
+        setback : int
+            Setback distance in meters.
+
+        Returns
+        -------
+        setbacks : list
+            List of setback geometries.
+        """
+        logger.debug('- Computing setbacks for county FIPS {}'
+                     .format(cnty.iloc[0]['FIPS']))
+        log_mem(logger)
+        mask = features.centroid.within(cnty['geometry'].values[0])
+        tmp = features.loc[mask]
+
+        setbacks = [
+            (geom, 1) for geom in tmp.buffer(0).difference(
+                tmp.buffer(-1 * setback)
+            )
+        ]
+
+        return setbacks
+
+    def _parse_regulations(self, regulations_fpath):
+        """
+        Parse parcel regulations, reduce table to just property lines
+
+        Parameters
+        ----------
+        regulations_fpath : str
+            Path to parcel regulations .csv file
+
+        Returns
+        -------
+        regulations : pandas.DataFrame
+            Parcel regulations table
+        """
+        regulations = super()._parse_regulations(regulations_fpath)
+
+        mask = regulations['Feature Type'].apply(str.strip) == 'Property Line'
+        regulations = regulations.loc[mask]
+
+        return regulations
+
+    def _check_regulations(self, features_fpath):
+        """
+        Reduce regs to state corresponding to features_fpath if needed.
+
+        Parameters
+        ----------
+        features_fpath : str
+            Path to shape file with features to compute setbacks from.
```

Comment:
Needs to have the state in the filename?

Reply:
Yup, added that to docstring!